### ISSUE TITLE
Potential bug in gradient_boosting.py

### ISSUE THREAD

**Issue by rminhas** [TorchEnsemble-Community/Ensemble-Pytorch#98]
It seems that the `fit()` method of the `_BaseGradientBoosting` class always uses `nn.MSELoss()` even when doing classification. Since the `self._criterion` variable is unused, the choice of the appropriate loss function needs be made within `_BaseGradientBoosting` (rather than `GradientBoostingClassifier` or `GradientBoostingRegressor`). Currently, that doesn't seem to be happening. Of course, it would be better if the `fit()` method in `_BaseGradientBoosting` class would use `self._criterion` instead of defining a local `criterion` variable.

Here is the relevant snippet of the `_BaseGradientBoosting` class:

```python
class _BaseGradientBoosting(BaseModule):
...
    def fit(self, train_loader, epochs=100, use_reduction_sum=True, log_interval=100, test_loader=None,
               early_stopping_rounds=2, save_model=True, save_dir=None):
        # Instantiate base estimators and set attributes
        for _ in range(self.n_estimators):
            self.estimators_.append(self._make_estimator())
        self._validate_parameters(epochs, log_interval, early_stopping_rounds)
        self.n_outputs = self._decide_n_outputs(train_loader)

        # Utils
        criterion = (
            nn.MSELoss(reduction="sum") if use_reduction_sum else nn.MSELoss()
        )
...
```
